packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift

```diff
@@ -73,11 +73,15 @@ extension SwiftAdobeAnalyticsPlugin {
 extension SwiftAdobeAnalyticsPlugin: AdobeAnalyticsProtocol {
   
   public func trackAction(_ action: String, data: [String : String]?) {
-    ACPCore.trackAction(action, data: data)
+    if (UIApplication.shared.applicationState != .background) {
```

Comment:
I'd rather go for a Swiftier version, maybe :
`open func shouldTrackAction() -> bool {
    return UIApplication.shared.applicationState != .background
}`
then
`guard shouldTrackAction() else {
    return
}
ACPCore.trackAction(action, data: data)`

Reply:
Not gonna happen... 
No way that's more readable

Reply:
Extracting the if check to a method is arguable.
but then it should probably be something like
`if (isSafeToTrack()) track()` or `if (notInBackground()) track()`
because `if (shouldTrack()) track()` makes it so you have to go to the implementation to know the behaviour, while it's actually relevant.

But `UIApplication.shared.applicationState != .background` is very readable, leaving it like that.